### PR TITLE
Fixes #12761: Add the possibility to customize filters and tests for jinja2 templating

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/12761

Replacing previous PR: https://github.com/Normation/ncf/pull/769

### CODE REVIEW

File: tree/10_ncf_internals/modules/templates/jinja2-templating.py [hunk 1]
```diff
@@ -67,6 +74,24 @@ def render(opts, args):
     if opts.strict:
         env.undefined = StrictUndefined
 
+    # Register customs
+    sys.path.append('/var/rudder/ncf/local/10_ncf_internals/modules/templates/')
+
+    if pkgutil.find_loader('jinja2_custom') is None:
+        custom_filters = False
+    else:
+        custom_filters = True
```
Comment (peckpeck) [inline tree/10_ncf_internals/modules/templates/jinja2-templating.py]: this is one line: 
custom_filters = pkgutil.find_loader('jinja2_custom') is not None

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
OK, merging this PR

File: tree/30_generic_methods/file_from_template_jinja2.cf [hunk 2]
```diff
@@ -52,6 +52,9 @@
 # Note: You can add comments in the template, that will not be rendered in the output file with
 # `{# ... #}`.
 #
+# You can extend the Jinja2 templating engine by adding custom FILTERS and TESTS in the script
```
Comment (amousset) [inline tree/30_generic_methods/file_from_template_jinja2.cf]: We need an example of the content of such a file.

**Pull request merged**